### PR TITLE
Batch UI

### PR DESCRIPTION
**Pull request by aravintsamy-converse**
this branch have batch for student and teacher batch creation with day and time durations

### CODE REVIEW

File: app/controllers/admin/batches_controller.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          ]
+        end
+        format.json { render json: @batch.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def show
+    respond_to do |format|
+      format.turbo_stream
```
Comment (aravindhan-converse) [inline app/controllers/admin/batches_controller.rb:35]: missing error handline

File: app/controllers/admin/batches_controller.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  def create
+    @batch = Batch.new(batch_params)    
+    respond_to do |format|
+      if @batch.save   
+        @batches = Batch.get_batches(params)
+        format.turbo_stream
+      else
+        format.turbo_stream do
+          render turbo_stream: [
+            turbo_stream.replace('batch-admin-form', partial: 'admin/batches/form', locals: { batch: @batch, batch_timings: @batch.batch_timings })
```
Comment (aravindhan-converse) [inline app/controllers/admin/batches_controller.rb:25]: notification message is missing for notify the error to uesr

File: app/controllers/admin/batches_controller.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  def update
+    respond_to do |format|
+      if @batch.update(batch_params)
+        @batches = Batch.get_batches(params)
+        format.turbo_stream
+        format.json { render :show, status: :ok, location: admin_batch_url(@batch) }
+      else
+        format.turbo_stream do
+          render turbo_stream: turbo_stream.update('edit-batch-popup', partial: 'admin/batches/edit',
+                                                                        locals: { batch: @batch })
```
Comment (aravindhan-converse) [inline app/controllers/admin/batches_controller.rb:57]: same here notify msg is missing

File: app/controllers/admin/batches_controller.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                                                                  locals: { message: 'Batch deletion failed.', type: 'notice' })
+        end
+      end
+    end
+  end
+
+  private
+
+  def set_batch
+    @batch = Batch.find(params[:id])
```
Comment (aravindhan-converse) [inline app/controllers/admin/batches_controller.rb:87]: use find_by

File: app/models/batch.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  validates :effective_from, presence: true
+  validates :effective_to, presence: true
+  validates :primary_trainer_id, presence: true
+  validates :secondary_trainer_id, presence: true
+
+  def self.get_batches(params)
+    page = (params[:page].presence&.to_i&.positive? ? params[:page].to_i : 1)
+    per_page = [(params[:per_page].presence&.to_i || 10).to_i, 1].max
+    Batch.includes(:course).search_by_name(params[:search]).search_by_course(params[:course]).paginate(page: page, per_page: per_page)
+
```
Comment (aravindhan-converse) [inline app/models/batch.rb:20]: check alignment

File: app/controllers/admin/batches_controller.rb (hunk 1 above)
Comment (aravintsamy-converse) [reply app/controllers/admin/batches_controller.rb:35]: error handled done

File: app/controllers/admin/batches_controller.rb (hunk 2 above)
Comment (aravintsamy-converse) [reply app/controllers/admin/batches_controller.rb:25]: added

File: app/models/batch.rb (hunk 5 above)
Comment (aravintsamy-converse) [reply app/models/batch.rb:20]: done

File: app/controllers/admin/batches_controller.rb (hunk 4 above)
Comment (aravintsamy-converse) [reply app/controllers/admin/batches_controller.rb:87]: done

**Pull request merged**